src/dsql/ExprNodes.cpp

```diff
@@ -3465,8 +3473,16 @@ bool CastNode::setParameterType(DsqlCompilerScratch* /*dsqlScratch*/,
 // Generate BLR for a data-type cast operation.
 void CastNode::genBlr(DsqlCompilerScratch* dsqlScratch)
 {
-	dsqlScratch->appendUChar(blr_cast);
+	if (format.hasData())
+	{
+		dsqlScratch->appendUChar(blr_cast_format);
+		dsqlScratch->appendString(0, format);
```

Comment:
You are ignoring the (implicit or explicit) connection charset (not `_WIN1252` in the example):
```
SELECT CAST(localTIMESTAMP AS VARCHAR(45) FORMAT _WIN1252 'DD.MM.YEAR HH24:MI:SS "is" J "Julian day"') FROM RDB$DATABASE;
```